admin/commands/storage/read_blocks.go

```diff
@@ -0,0 +1,92 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/admin"
+	"github.com/onflow/flow-go/admin/commands"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+)
+
+var _ commands.AdminCommand = (*ReadBlocksCommand)(nil)
+
+type readBlocksRequest struct {
+	blocksRequest    *blocksRequest
+	numBlocksToQuery uint64
+}
+
+type ReadBlocksCommand struct {
+	state  protocol.State
+	blocks storage.Blocks
+}
+
+func (r *ReadBlocksCommand) Handler(ctx context.Context, req *admin.CommandRequest) (interface{}, error) {
+	data := req.ValidatorData.(*readBlocksRequest)
+	var result []*flow.Block
+	var blockID flow.Identifier
+
+	if header, err := getBlockHeader(r.state, data.blocksRequest); err != nil {
+		return nil, fmt.Errorf("failed to get block header: %w", err)
+	} else {
+		blockID = header.ID()
+	}
+
+	for i := uint64(0); i < data.numBlocksToQuery; i++ {
+		block, err := r.blocks.ByID(blockID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get block by ID: %w", err)
+		}
+		result = append(result, block)
+		if block.Header.Height == 0 {
+			break
+		}
+		blockID = block.Header.ParentID
+	}
+
+	return convertToInterfaceList(result)
+}
+
+func (r *ReadBlocksCommand) Validator(req *admin.CommandRequest) error {
+	input, ok := req.Data.(map[string]interface{})
+	if !ok {
+		return errors.New("wrong input format: expected JSON")
+	}
```

Comment:
Should return sentinel errors for invalid request data to be asserted in the test, something like the following

```
var (
    ErrValidatorReqDataFormat = fmt.Errorf("wrong input format: expected JSON"
)
```

```suggestion
	if !ok {
		return ErrValidatorReqDataFormat
	}
```